utils/wikize_refs.py

```diff
@@ -74,12 +85,33 @@ def parse_args():
     parser.add_argument("-w", "--warn",
                       default=False,
                       action="store_true",
-                      help="Warn instead of error during (most) error checks.")
+                      help="Warn instead of exit during (most) error checks.")
+
+    parser.add_argument("-c", "--check-links",
+                      type=int, default=0,
+                      help="Specify a timeout>0 in seconds for checking for broken links.")
 
     parser.add_argument("-i", "--in-place",
                       default=False,
                       action="store_true",
-                      help="Modify input file in-place.")
+                      help="Modify input file in-place (destructive).")
+
+    parser.add_argument("-g", "--gather-linkdefs",
+                      default=False,
+                      action="store_true",
+                      help="Gather all link definitions to the end of the file (destructive).")
+
+    parser.add_argument("-l", "--linkdef-db",
+                      type=str, default=None,
+                      action="append",
+                      help="Specify name of file to use as a linkdef database from which \
+                            to resolve footnote references in the file being processed. Any \
+                            used linkdefs will be copied from that file to the file being \
+                            processed here. Multiple -l options are allowed.")
```

Comment:
Is this a version of a bibtex references file?  If so, that would be fantastic!  (I guess I need to look at the test to see if this is the case.)

Reply:
No...not bibtex but we could think of supporting that. Its just some other `.md` file that contains link definitions that are referenced in the *current* file.

Reply:
> No...not bibtex but we could think of supporting that. Its just some other .md file that contains link definitions that are referenced in the current file.

@markcmiller86, I don't mean bibtext formatting.  I mean the idea of a separate file that contains these citation reference defections.  That is all.  I will add a test to verify this is indeed the case.

Reply:
I added tests already ;)

Reply:
Are there any tests that demonstrate?

> Multiple -l options are allowed.

I can't seem to find any.  Would it be hard to put in a test that has two arguments `-l <db-file1> -l <db-file2>`?  I could give it a try but given the failure of the test `test wikize_refs_linkdef_databases_pure` from PR #967, I am not sure how to go about doing that since I don't know that I understand the constraints on these files.